src/routes/user/apps/webhooks/WebhooksRouter.ts

```diff
@@ -46,27 +46,32 @@ function getPushedBranches(req: express.Request) {
 }
 
 router.post('/triggerbuild', urlencodedParser, function(req, res, next) {
+    const extracted = InjectionExtractor.extractAppAndUserForWebhook(
+        res
+    )
+    const { serviceManager, namespace } = extracted.user
+    const { appName, app } = extracted
+
+    if (!app || !serviceManager || !namespace || !appName) {
+        Logger.e(new Error(
+            'Something went wrong during trigger build. Cannot extract app information from the payload.'
+        ))
+        res.send(
+            new BaseApi(
+                ApiStatusCodes.STATUS_ERROR_GENERIC,
+                'Error triggering a build'
+            )
+        )
+        return
+    }
```

Comment:
Did we have to move this block outside of the promise block? 

The benefit of keeping it in the promise block is that, on top of being consistent with everything else, if there is any error occurs, we'll go to the catch block and emit the a proper API error and log the error properly, rather than letting express js throw a 5xx error.

Reply:
The problem was the sending of the response of success was not in a promise and I needed mine above it.  I can restructure the function to return a promise and then put the success in there.

Reply:
This looks good. But the second part of my comment is not addressed yet. Sorry if it was confusing... My concern was that if `InjectionExtractor.extractAppAndUserForWebhook` throws an exception, then we won't go to the catch block that you've added. Instead, it goes to the generic express js error handler and a 500 error will be thrown. My suggestion is to bring these lines:
```
    const extracted = InjectionExtractor.extractAppAndUserForWebhook(
        res
    )
    const { serviceManager, namespace } = extracted.user
    const { appName, app } = extracted
```

inside the promise. Of course, we'll have to change them from `const` to `let` and define them above the promise and assign the values inside the promise.